lib/engine/game/g_18_usa/step/scrap_train.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative 'scrap_train_module'
+module Engine
+  module Game
+    module G18USA
+      module Step
+        class ScrapTrain < Engine::Step::Base
+          include ScrapTrainModule
+          def actions(entity)
+            actions = []
+            actions << 'scrap_train' if entity == current_entity && entity.corporation? &&
+                entity.trains.any? { |t| @game.pullman_train?(t) }
+            actions << 'pass' if blocks?
+            actions
+          end
+
+          def blocks?
+            @round.paid_loans[current_entity] && can_scrap_train?(current_entity)
```

Comment:
Hmm...we don't want both repay loan and scrap trains to block, so that you have to pass both. Can you make this step before repay loans and then rely on it blocking for the player to get a final opportunity before their turn ends?

Reply:
Instead I included ScrapTrainModule into 18USA::Step::Loan and removed the ScrapTrain step from the operating round so that the Loan step can be the only blocking step - if the company can do a loan or scrapping action